kiwi/filesystem/isofs.py

```diff
@@ -36,28 +37,39 @@ def create_on_file(self, filename, label=None, exclude=None):
         :param string label: unused
         :param string exclude: unused
         """
+        iso_tool = IsoToolsCdrTools(self.root_dir)
+        iso_tool_name = iso_tool.get_tool_name()
+
         iso = Iso(self.root_dir)
-        iso.init_iso_creation_parameters(
-            self.custom_args['create_options']
+        iso.setup_isolinux_boot_path()
+
+        iso.create_header_end_marker()
+
+        iso_tool.init_iso_creation_parameters(
+            iso.create_sortfile(), self.custom_args['create_options']
         )
-        iso.add_efi_loader_parameters()
+        iso_tool.add_efi_loader_parameters()
+
         Command.run(
             [
-                Iso.get_iso_creation_tool()
-            ] + iso.get_iso_creation_parameters() + [
+                iso_tool_name
+            ] + iso_tool.get_iso_creation_parameters() + [
                 '-o', filename, self.root_dir
```

Comment:
This option handling needs to be addressed as we can't assume this to be correct for any tool

Reply:
I guess this comment is in line to my comment below, or you meant something different?